### ISSUE TITLE
regression --connection local is not honored by falcon/roles/falcon_install/tasks/auth.yml

### ISSUE THREAD

**Issue by TyraelTLK** [CrowdStrike/ansible_collection_falcon#305]
version 3.2.27 introduced this issue.
When running the role on localhost using the local connection it tries anyway to use ssh:

`ansible-playbook --connection=local -i localhost, playbook.yml
`
first run:

```
TASK [crowdstrike.falcon.falcon_install : ansible.builtin.include_tasks] *******
included: /usr/share/ansible/collections/ansible_collections/crowdstrike/falcon/roles/falcon_install/tasks/auth.yml for localhost

TASK [crowdstrike.falcon.falcon_install : CrowdStrike Falcon | Authenticate to CrowdStrike API] ***
The authenticity of host 'localhost (127.0.0.1)' can't be established.
ECDSA key fingerprint is SHA256:oXjS6QUwT6fH8qT98eJwmN7hqCMgv3XJgzDcpY33bhU.
Are you sure you want to continue connecting (yes/no/[fingerprint])?
```

second run:
```
TASK [crowdstrike.falcon.falcon_install : ansible.builtin.include_tasks] *******
included: /usr/share/ansible/collections/ansible_collections/crowdstrike/falcon/roles/falcon_install/tasks/auth.yml for localhost

TASK [crowdstrike.falcon.falcon_install : CrowdStrike Falcon | Authenticate to CrowdStrike API] ***
fatal: [localhost]: UNREACHABLE! => {"changed": false, "msg": "Failed to connect to the host via ssh: |       ***
\nroot@localhost: Permission denied (publickey,password).", "unreachable": true}

NO MORE HOSTS LEFT *************************************************************

PLAY RECAP *********************************************************************
localhost                  : ok=78   changed=9    unreachable=1    failed=0    skipped=53   rescued=1    ignored=3
```

**Comment by TyraelTLK** [comment]
I think that is related to https://github.com/CrowdStrike/ansible_collection_falcon/issues/289

**Comment by carlosmmatos** [comment]
@TyraelTLK this seems to be an issue with your environment. At some point, it seems like you might be overriding the `ansible_connection` variable to ssh. I would check out the [variable precedence](https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_variables.html#understanding-variable-precedence) docs to understand that passing via cli will be overridden pretty much anywhere else.

Here's an example playbook to demonstrate:
```yaml
---
- name: Test connection information with local connection
  hosts: localhost
  gather_facts: false
  tasks:
    - name: Print connection information
      ansible.builtin.debug:
        msg: "Connection information: {{ ansible_connection }}"
    - name: Simple command
      ansible.builtin.command: whoami

- name: Test connection information with ssh connection
  hosts: localhost
  gather_facts: false
  vars:
    ansible_connection: ssh
  tasks:
    - name: Print connection information
      ansible.builtin.debug:
        msg: "Connection information: {{ ansible_connection }}"
    - name: Simple command
      ansible.builtin.command: whoami
```

Here is what the run would look like:
```shell
PLAY [Test connection information with local connection] ************************************************************************************************************************************

TASK [Print connection information] *********************************************************************************************************************************************************
ok: [localhost] => {
    "msg": "Connection information: local"
}

TASK [Simple command] ***********************************************************************************************************************************************************************
changed: [localhost]

PLAY [Test connection information with ssh connection] **************************************************************************************************************************************

TASK [Print connection information] *********************************************************************************************************************************************************
ok: [localhost] => {
    "msg": "Connection information: ssh"
}

TASK [Simple command] ***********************************************************************************************************************************************************************
The authenticity of host 'localhost (127.0.0.1)' can't be established.
ED25519 key fingerprint is SHA256:g9BPcUuXNotRdTEbs+uWey9GNRJUNPQ3tvCMeVPrj1M.
This host key is known by the following other names/addresses:
    ~/.ssh/known_hosts:2: [hashed name]
Are you sure you want to continue connecting (yes/no/[fingerprint])? yes
fatal: [localhost]: UNREACHABLE! => {"changed": false, "msg": "Failed to connect to the host via ssh: Warning: Permanently added 'localhost' (ED25519) to the list of known hosts.\r\nexample@localhost: Permission denied (publickey,password).", "unreachable": true}

PLAY RECAP **********************************************************************************************************************************************************************************
localhost                  : ok=3    changed=1    unreachable=1    failed=0    skipped=0    rescued=0    ignored=0  
```

**Comment by TyraelTLK** [comment]
HI @carlosmmatos ,

here the output:

# ansible-playbook --connection=local -i localhost, playbook.yml
```
PLAY [Test connection information with local connection] *****************************************************************************************************

TASK [Print connection information] **************************************************************************************************************************
ok: [localhost] => {
    "msg": "Connection information: local"
}

TASK [Simple command] ****************************************************************************************************************************************
changed: [localhost]

PLAY [Test connection information with ssh connection] *******************************************************************************************************

TASK [Print connection information] **************************************************************************************************************************
ok: [localhost] => {
    "msg": "Connection information: ssh"
}

TASK [Simple command] ****************************************************************************************************************************************
fatal: [localhost]: UNREACHABLE! => {"changed": false, "msg": "Failed to connect to the host via ssh: |       *******************************************************************\n \nroot@localhost: Permission denied (publickey,password).", "unreachable": true}

PLAY RECAP ***************************************************************************************************************************************************
localhost                  : ok=3    changed=1    unreachable=1    failed=0    skipped=0    rescued=0    ignored=0
```

The previous tasks are correctly executed.
If I remove 3.2.27 and install 3.2.26 it runs correctly.

**Comment by carlosmmatos** [comment]
hmm... what version of Ansible are you running?

I wonder if this #300 issue is what you are experiencing. Are you targeting Linux host when it fails? Windows? Or a combo?

**Comment by carlosmmatos** [comment]
The delegate_to should only occur for non linux systems:
`delegate_to: "{{ 'localhost' if ansible_facts['system'] != 'Linux' else omit }}"`

Apparently there was a bug with the `omit` portion on certain versions of Ansible.. I think 2.12.x. 

Basically the diff between 3.2.27|26 for that line is:
```diff
- delegate_to: localhost
+ delegate_to: "{{ 'localhost' if ansible_system != 'Linux' else omit }}"
```

So before, everything was delegating to localhost - regardless of OS. Now it should only do that for windows/mac.. but I suppose if you were to have a mixed inventory, since this is a `run_once` task, it could choose linux and 'omit', making the bug above appear.

**Comment by TyraelTLK** [comment]
I'm running on localhost on Ubuntu 20.04

# ansible-playbook --version
ansible-playbook [core 2.12.10]
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/root/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3/dist-packages/ansible
  ansible collection location = /root/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/bin/ansible-playbook
  python version = 3.8.10 (default, Nov 14 2022, 12:59:47) [GCC 9.4.0]
  jinja version = 2.10.1
  libyaml = True
  
Let me try a different ansible version.
  
Passing -e "ansible_connection=local" works fine as a workaround.

**Comment by carlosmmatos** [comment]
@TyraelTLK can you try the following test collection with the new PR #306?
```yaml
collections:
  - name: https://github.com/carlosmmatos/ansible_collection_falcon.git
    type: git
    version: test-issue-with-delegate-to
```
or
`ansible-galaxy collection install git+https://github.com/carlosmmatos/ansible_collection_falcon.git,test-issue-with-delegate-to`

**Comment by TyraelTLK** [comment]
I'm not able to reproduce the issue on ubuntu **22.04** with **ansible 2.14**:
```
# ansible-playbook --version
ansible-playbook [core 2.14.2]
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/root/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3/dist-packages/ansible
  ansible collection location = /root/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/bin/ansible-playbook
  python version = 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0] (/usr/bin/python3)
  jinja version = 3.0.3
  libyaml = True
```

I tried also to reproduce on Ubuntu **20.04** with **ansible-core-2.13.7** installed with pip and I can't reproduce it.

I tested ansible-galaxy collection install git+https://github.com/carlosmmatos/ansible_collection_falcon.git,test-issue-with-delegate-to on **Ubuntu 20.04** with **ansible 2.12**  and it is now working fine.

**Comment by carlosmmatos** [comment]
@TyraelTLK thank you for troubleshooting and testing those versions! So I guess that Ansible bug with omitting was causing the issue in 2.12. Not a big deal, and since we support Ansible from version 2.11 in our collection, I will be getting the current PR merged, along with a new release 3.2.28 soon.

**Comment by carlosmmatos** [comment]
@TyraelTLK keep an eye out on Galaxy for latest release.
